backend/src/main/java/org/eclipse/tractusx/semantics/registry/service/ShellService.java

```diff
@@ -81,17 +96,63 @@ public Shell findShellByExternalId(String externalShellId) {
     }
 
     @Transactional(readOnly = true)
-    public ShellCollectionDto findAllShells(int page, int pageSize) {
-        Pageable pageable = PageRequest.of(page, pageSize, Sort.Direction.ASC, "createdDate");
-        Page<Shell> shellsPage = filterSpecificAssetIdsByTenantId(shellRepository.findAll(pageable));
-        return ShellCollectionDto.builder()
-                .currentPage(pageable.getPageNumber())
-                .totalItems((int) shellsPage.getTotalElements())
-                .totalPages(shellsPage.getTotalPages())
-                .itemCount(shellsPage.getNumberOfElements())
-                .items(shellsPage.getContent())
-                .build();
-    }
+    public ShellCollectionDto findAllShells(Integer pageSize,String cursorVal) {
+
+      pageSize = getPageSize( pageSize );
+      ShellCursor cursor = new ShellCursor( pageSize, cursorVal );
+      var specification = new ShellSpecification<Shell>( SORT_FIELD_NAME, cursor );
+
+      Page<Shell> shellPage = filterSpecificAssetIdsByTenantId( shellRepository.findAll( specification, ofSize( cursor.getRecordSize() ) ) );
+      var shellsPage = shellPage.getContent();
+
+      String nextCursor=null;
+
+      if(shellsPage.size()>0) {
+         nextCursor = cursor.getEncodedCursor(
+               shellsPage.get( shellsPage.size() - 1 ).getCreatedDate(),
+               shellPage.hasNext() );
+      }
+
+      return ShellCollectionDto.builder()
+            .items( shellsPage )
+            .cursor( nextCursor )
+            .build();
+   }
+
+   @Transactional( readOnly = true )
+   public SubmodelCollectionDto findAllSubmodel( Integer pageSize, String cursorVal, Shell assetID ) {
+      pageSize = getPageSize( pageSize );
+
+      ShellCursor cursor = new ShellCursor( pageSize, cursorVal );
+      var specification = new ShellSpecification<Submodel>( SORT_FIELD_NAME, cursor );
+      Page<Submodel> shellPage = submodelRepository.findAll( Specification.allOf( hasShellFkId( assetID ).and( specification ) ),
+            ofSize( cursor.getRecordSize() ) );
+
+      var shellsPage = shellPage.getContent();
+      String nextCursor=null;
+
+      if(shellsPage.size()>0) {
+         nextCursor = cursor.getEncodedCursor(
+               shellsPage.get( shellsPage.size() - 1 ).getCreatedDate(),
+               shellPage.hasNext()
+         );
+      }
+      return SubmodelCollectionDto.builder()
+            .items( shellsPage )
+            .cursor( nextCursor )
+            .build();
+   }
+
+   private static Integer getPageSize( Integer pageSize ) {
```

Comment:
Is there an reason to make this method static? 
You assign new value to pageSize -> Then you can also use void and change the name like validateAndSetPageSize or something like that. But it is up to you :)